### ISSUE TITLE
Contacts app should create human readable vcf-file names when exporting a contact record.

### ISSUE THREAD

**Issue by j-ed** [nextcloud/server#2880]
Due to the fact that this is a SabreDAV issue I was told to open a new issue ticket here.
(https://github.com/nextcloud/contacts/issues/68)

## Expected behavior
The Contacts app should create a human readable vcf-file name when exporting a contact record.
E.g. if the full name of a contact is `Tom Johnes` the vcf-file name should be `tom_johnes.vcf` and not `6d9b3192-4eb4-4f6b-81d8-971ddcb6b28f.vcf`.

## Current behavior
The Contacts app uses the UID of a record to create the file name of the vcf-file, like `6d9b3192-4eb4-4f6b-81d8-971ddcb6b28f.vcf`.

### Steps to reproduce
1. Open the Contacts app and export a contact record.
2. Check-out the default vcf-file name.

#### Server Configuration

OS: Linux 3.2.82
Web server: Apache2 2.4.25
Database: MariaDB 5.5.53
PHP version: 5.6.23
Nextcloud version: 10.0.2
Contacts app: 1.5.2
#### Client Configuration

Browser: Mozilla Firefox 50.1.0
Operating system: Windows 7

**Comment by MorrisJobke** [comment]
cc @blizzz

**Comment by blizzz** [comment]
With Nextcloud 11, the DAV library is not the issue, the necessary feature for native exports got was added in 3.2 which is used in Nc 11: https://github.com/fruux/sabre-dav/pull/814

But the contacts app is saving the contacts card already using the ID and also builds the download link as such. Passing back to contacts, sorry…

**Comment by blizzz** [comment]
Too early. The linked PR does it only for the addressbook, single contacts are not affected by this. This needs to be solved in core (or upstream) indeed. Basically, on the get request the DAV stuff should add a content disposition header, perhaps only when ``?export`` is appended to the URL. Perhaps we can just extend the ExportPlugin from Sabre, alas I am not convinced it is a 100% generic solution to force a name mode of "FN" over the original file name. Could make other things explode ;)

**Comment by blizzz** [comment]
The server part is here: https://github.com/nextcloud/server/pull/2940

However contacts will need to add an  "?export" to the download URL once this is merged.

**Comment by blizzz** [comment]
Fixed by https://github.com/nextcloud/contacts/issues/83
